### PR TITLE
fix missing/malformed cloud event

### PR DESCRIPTION
**Pull request by lillianzhang331**
When a Function is invoked, the HTTP request headers contain metadata in the Cloud Event format ().
The invokers parse these headers, and use the metadata within to set the various function context objects.
The Functions requests contain a number of custom fields using "extensions" to the Cloud Event base spec.

Currently, the Node.js invoker errors, and returns a slightly unhelpful error message.

For example, if the extensions are missing entirely, the function fails with:
"Could not parse CloudEvent: Cannot read property 'toString' of undefined"

Or if the extensions contain malformed JSON, the function fails with:
"Could not parse CloudEvent: Unexpected end of JSON input"

Fix is to handle these cases and ensure the function can still run, but with empty/null/... salesforce contexts. For malformed JSON, an error will be thrown with "Unexpected end of JSON input" as the error message.

GUS-W-9654393.

### CODE REVIEW

File: src/cloud-event.ts [hunk 1]
```diff
@@ -53,5 +58,9 @@ export interface SalesforceFunctionContext {
 }
 
 function parseBase64Json<A>(data: string): A {
-  return JSON.parse(Buffer.from(data, "base64").toString("utf-8"));
+  try {
+    return JSON.parse(Buffer.from(data, "base64").toString("utf-8"));
+  } catch (error) {
+    throw new Error(error.message);
```
Comment (joshwlewis) [inline src/cloud-event.ts:64]: Could we add some helpful context here? I think this is just going to show a `Unexpected end of JSON input` type message which won't give the developer much of a clue to the problem. Maybe something more like ``Could not execute function. Function arguments could not be determined due to invalid JSON body: ${error.message}``.

**Review by joshwlewis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Automatically expand ReferenceIDs with `.toApiString()` ([#133](https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/133))
 - Add unit testing for src/index.ts ([#118](https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/118))
 - Utilize @salesforce/core for logging with logfmt ([#119](https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/119))
+- Handle malformed/missing cloud event salesforce extension ([#141](https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/141))s
```
Comment (schneems) [inline CHANGELOG.md:14]: Stray `s` at the end here. 

```suggestion
- Handle malformed/missing cloud event salesforce extension ([#141](https://github.com/forcedotcom/sf-fx-runtime-nodejs/pull/141))
```

Generally I put new changelog entries at the top so it ends up being in "timeline" order. That way if someone was using main and their stuff broke the first entry on the top is the most likely candidate. I don't think you need to move it, but wanted to leave you a note.

**Review by schneems — COMMENTED** [review]
Made a note on the changelog format. Overall change makes sense to me. The tests are mostly kept intact and comments/skip removed. I agree with josh on giving a little love to the error messaging. Thanks for working on this!

File: test/cloud-event.ts [hunk 3]
```diff
@@ -89,21 +80,15 @@ describe("parseCloudEvent", () => {
...
-    );
 
-    expect(cloudEvent.id).to.equal(inputEvent.id);
-    // Should this even be `null`, or something else?
-    expect(sfContext).to.equal(null);
-    expect(sfFunctionContext).to.equal(null);
+    try {
+      parseCloudEvent(headers, body);
+    } catch (error) {
+      expect(error.toString()).to.include("Unexpected end of JSON input");
```
Comment (joshwlewis) [inline test/cloud-event.ts:87]: I think the new error messaging you added is more important to check for. Maybe add an assertion about it?
```suggestion
      expect(error.toString()).to.include("Function arguments could not be determined due to invalid JSON");
      expect(error.toString()).to.include("Unexpected end of JSON input");
```

**Review by joshwlewis — APPROVED** [review]

**Pull request merged**